zebra/zebra_rib.c

```diff
@@ -1503,9 +1507,15 @@ static void rib_process_update_fib(struct zebra_vrf *zvrf,
 		 * interface up before IPv4 or IPv6 protocol is ready
 		 * to add routes.
 		 */
-		if (!CHECK_FLAG(new->status, ROUTE_ENTRY_INSTALLED) ||
-		    RIB_SYSTEM_ROUTE(new))
+		if (!CHECK_FLAG(new->status, ROUTE_ENTRY_INSTALLED))
 			rib_install_kernel(rn, new, NULL);
+		else {
```

Comment:
is this else even needed?

Reply:
the thought here was "for a _normal_ route, that needs to be installed, call through and install it. else, for a system route that already has INSTALLED set, ensure that the nexthops and so forth are correctly set." What appeared to be happening was:

RIB_SYSTEM_ROUTE check
rib_install_kernel()
no-op dplane update
dplane results processing in zebra_rib.c sees system route, and calls fixup_system()

so I was trying to get fixup_system() called without the loop through the dataplane.

Reply:
from our discussion the other week I thought you were trying to solve a different problem, namely that we were calling in multiple times.  If we have already called in what do we need to set?  nexthop_active_udpate already correctly sets the nexthop flags right?

Reply:
No, active_update() twiddles the ACTIVE flag, but not the FIB flag. without a trip through fixup_system(), to work the FIB flags, the system routes appear to be not installed. the dplane results process does end up there, but that's sort of a side-effect, and it seems clearer to just ... fixup the system routes right away.

Reply:
I intentionally put everything through the data plane to remove extra code and this just adds it back in.  Is there a case where we are not sending the system route through and it is not getting marked appropriately?

Reply:
The routes are going through the dataplane. I think the case is that the work done to push the system routes through is entirely to get the 'fixup_system' api called during results processing. So it seemed like maybe just ... calling 'fixup' on them would save that work. The system routes become a special case that every dataplane module has to anticipate - that sort of just moves the work, the special case, from zebra into another layer, and I'm not sure that's a net win.

Reply:
The move of this to the data plane *greatly* simplified zebra route processing.  So much so it was far easier to understand.  I'm not interested in going backwards here.  NAK from me